diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index af4754a35e66..bc49ebaf8a87 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -139,7 +139,7 @@ elif [ "${LOCALVERSION+set}" != "set" ]; then
 	# If the variable LOCALVERSION is set (including being set
 	# to an empty string), we don't want to append a plus sign.
 	scm=$(scm_version --short)
-	res="$res${scm:++}"
+	# res="$res${scm:++}"
 fi
 
 echo "$res"
